src/Services/FHIR/Observation/FhirObservationLaboratoryService.php

```diff
@@ -257,6 +260,29 @@ public function parseOpenEMRRecord($dataRecord = array(), $encode = false)
             $observation->setSubject(UtilsService::createRelativeReference("Patient", $dataRecord['patient']['uuid']));
         }
 
+        // Add specimen information (USCDI v5 requirement)
+        if (!empty($dataRecord['specimen_code'])) {
+            $specimen = new FHIRReference();
```

Comment:
OpenEMR\Services\FHIR\UtilsService::createRelativeReference() can be used here.  You're missing the Reference here though which needs to point to the uuid.  IE setReference('Specimen/<uuid>');